akka-http-core/src/main/scala/akka/http/javadsl/settings/Http2ServerSettings.scala

```diff
@@ -16,6 +16,9 @@ trait Http2ServerSettings { self: scaladsl.settings.Http2ServerSettings ⇒
 
   def getIncomingStreamLevelBufferSize: Int = incomingStreamLevelBufferSize
   def withIncomingStreamLevelBufferSize(newIncomingStreamLevelBufferSize: Int): Http2ServerSettings
+
+  def getMaxConcurrentStreams: Int = maxConcurrentStreams
+  def withMaxConcurrentStreams(newValue: Int): Http2ServerSettings
```

Comment:
One thought, I don't remember the exact HTTP/2 details, but, isn't there a way this should be sent to the client to inform it about how many sub streams it may use to the server?

Reply:
Hm indeed it can be set as SETTING frame:

```
5.1.2 Stream Concurrency

A peer can limit the number of concurrently active streams using the SETTINGS_MAX_CONCURRENT_STREAMS parameter (see Section 6.5.2) within a SETTINGS frame. The maximum concurrent streams setting is specific to each endpoint and applies only to the peer that receives the setting. That is, clients specify the maximum number of concurrent streams the server can initiate, and servers specify the maximum number of concurrent streams the client can initiate.

```

I guess this is more like a default value for us for now hm... The setting we don't respect yet I guess

Reply:
That's correct, in future we could announce and enforce this value with HTTP/2.